### DISCUSSION TITLE
Disable `QueryComplexity` rule when introspecting

### DISCUSSION THREAD

**Discussion by ruudk** [webonyx/graphql-php#1203] [Q&A] [answered]
I'm creating a new GraphQL API and I want to keep the query complexity as low as possible.

So I set it to 50 to see when I need to increase it.

But pretty soon I noticed that introspection was no longer working in GraphiQL because the introspection query is pretty complex.

Is there a way to only enforce complexity checks when the query is not an introspection query?

**Comment by spawnia** [comment, marked as answer]
Such an option does not exist currently, but it would make sense to me. In terms of implementation, we could simply not count introspection fields. That way, a query mixing introspection and other fields can still be rightfully rejected.

Feel free to create an issue, or even better a pull request 😉

**Reply by ruudk** [reply]
I tried to fix it but couldn't get it to work, so I created the issue here: https://github.com/webonyx/graphql-php/issues/1204
